php-checks/src/main/java/org/sonar/php/checks/HardCodedCredentialsInFunctionCallsCheck.java

```diff
@@ -0,0 +1,223 @@
+    String methodName = "__construct";
+    checkForSensitiveMethod(functionCallTree, fqn + "::" + methodName);
+
+    super.visitNewExpression(tree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    ExpressionTree callee = tree.callee();
+
+    if (callee.is(Kind.OBJECT_MEMBER_ACCESS)) {
+      // Is it even possible?
+      // TODO: Remove after more testing
+      throw new RuntimeException("Delete me after testing!");
+    }
+
+    if (callee.is(Kind.CLASS_MEMBER_ACCESS)) {
+      MemberAccessTree memberAccessTreeCallee = (MemberAccessTree) callee;
+      // TODO: Use Nils method
+      // TODO: How will this work, if method is not static? -> Test me!
+      String fqn = ((ClassNamespaceNameTreeImpl) memberAccessTreeCallee.object()).symbol().qualifiedName().toString();
+
+      String methodName = memberAccessTreeCallee.member().toString();
+      checkForSensitiveMethod(tree, fqn + "::" + methodName);
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private void checkForSensitiveMethod(FunctionCallTree tree, String fqnMethodName) {
+    if (SENSITIVE_FUNCTIONS.containsKey(fqnMethodName)) {
+      SensitiveMethod sensitiveMethod = SENSITIVE_FUNCTIONS.get(fqnMethodName);
+      ArgumentMatcher[] methodMatchers = sensitiveMethod.getCorrespondingMatchers();
+
+      String lowerCaseFunctionName = CheckUtils.getLowerCaseFunctionName(tree);
+      if (lowerCaseFunctionName == null) {
+        return;
+      }
+      checkArgument(tree, lowerCaseFunctionName, methodMatchers);
+    }
+  }
+
+  @Override
+  protected void createIssue(ExpressionTree argument) {
+    context().newIssue(this, argument, MESSAGE);
+  }
+
+  private static class SensitiveMethod {
+    private final String name;
+    private final String cls;
+
+    //TODO: Check if this is used somehow?
+    private final String type;
+    private final Set<Integer> sensitiveIndices;
+
+    private final List<String> orderedArguments;
+
+    public SensitiveMethod(String name, String cls, String type, Set<Integer> sensitiveIndices, List<String> orderedArguments) {
+      this.name = name;
+      this.cls = cls;
+      this.type = type;
+      this.sensitiveIndices = sensitiveIndices;
+      this.orderedArguments = orderedArguments;
+    }
+
+    public String uniqueName() {
+      return cls + "::" + name;
+    }
+
+    public ArgumentMatcher[] getCorrespondingMatchers() {
+      Function<ExpressionTree, Boolean> isRegularStringLiteral = tree -> tree.is(Kind.REGULAR_STRING_LITERAL);
+
+      ArgumentMatcher[] matchers = new ArgumentMatcher[sensitiveIndices.size()];
+
+      Iterator<Integer> iterator = sensitiveIndices.iterator();
+      int matcherIndex = 0;
+      while (iterator.hasNext()) {
+        Integer index = iterator.next();
+        matchers[matcherIndex] = new ArgumentVerifierUnaryFunction(index, orderedArguments.get(index), isRegularStringLiteral);
+        matcherIndex++;
+      }
+
+      return matchers;
```

Comment:
To avoid creating the same matcher multiple times. I would use a map combined with `computeIfAbsence`. This may reduce the number of objects we have to create.

Reply:
I agree that this could potentially reduce the # of matchers. I would have to base the key of the map on two variables (index:int and name:string). Do you have a better solution that doing some String magic?
String magic like: Key = <index>;<name>